common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -4756,6 +4756,10 @@ public static enum ConfVars {
         "Class to use for calculating available slots during split generation"),
     HIVE_TEZ_GENERATE_CONSISTENT_SPLITS("hive.tez.input.generate.consistent.splits", true,
         "Whether to generate consistent split locations when generating splits in the AM"),
+    HIVE_TEZ_SPLIT_FS_SERIALIZATION_THRESHOLD("hive.tez.split.fs.serialization.threshold", 524288,
```

Comment:
should this be split size config or global input config? we might have huge number of split causing OOM

Reply:
right, I was thinking about the same

in case of a global threshold, we'll serialize every split to fs once we reach that regardless of the size, so we might end up serializing smaller splits also (20-30Kb), but that's fine as we remain stable...we need to figure out a reasonable default, e.g. 256MB?

-1: disabled
0: serialize everything to fs
n: global threshold

Reply:
have no idea about the default size, you have more experience here

Reply:
okay, I'm setting it to 256MB, which is reasonable for every AM sizes between the typical 2-4GB range